source/common/http/http1/codec_impl.cc

```diff
@@ -241,6 +245,15 @@ void ResponseStreamEncoderImpl::encodeHeaders(const HeaderMap& headers, bool end
   connection_.addCharToBuffer('\r');
   connection_.addCharToBuffer('\n');
 
+  if (numeric_status == 204 || numeric_status < 200) {
+    // Per https://tools.ietf.org/html/rfc7230#section-3.3.2
+    setIsContentLengthAllowed(false);
+  } else {
+    // Make sure that if we encodeHeaders(100) then encodeHeaders(200) that we
```

Comment:
OK thanks for the comment, this was totally not intuitive. Can you potentially add a unit test specific to this case?

Reply:
Happy to.  We already have an end to end test for this case which I tripped in my first pass.

Also fun if you make the silly mistake of skipping the entire branch instead of just the append, you end up with chunk_encoding_ = true on 204s-with-no-body which for some reason rather than appending a 0 byte chunked body caused inexplicable seg-faults.

It's what made me bump this from low "it's a 5 line change affecting 204 HTTP/1.1 responses" to medium "man the HCM is complicated"  On third thought I'm going to bump it all the way to high risk because man the HCM is complicated.